Menu js

xandro2021 opened the pull request:
In this PR:
+ Install and Set JavaScript Linters
+ Separate some SVG images from the HTML document to separate files in order to make it easier to work in today's assignation (HTML cleaner)
+ Create the js folder and file for today assignation with the initial idea about how to approach the requirements
+ Add a close icon to be toggled with the hamburger icon 
+ Style the menu navbar when the hamburger bottom is pressed
+ Add navigation to the different sections required by the assignation
+ Pulish the initial javascript file once it was created the styles for the menu when it is triggered by the hamburger icon click/touch
+ Add smooth scroll to the document
+ Separate my own work from the file that reset the browser's defaults styles
+ Ensure that the recent functionality only works for small screens (less than 758px)

Thank you for your review :) It was hard to come to this point, got lost in my thoughts and my coding partner's thoughts but somehow here I got it.

xandro2021 committed:
Install and Set JavaScript Linters

Ruthmy committed:
Changed some svg imgs

Ruthmy committed:
Added the js folder and the menu.js file for mobile

Ruthmy committed:
Added auxiliary classes for javascript

xandro2021 committed:
Add close icon for substitution of the Hamburger icon

xandro2021 committed:
Style how the menu will look when the Hamburger is active

+ It was given the sticky position and a higher z-index to the header,
in order to make easier the navigation by having available the navbar
+ The display of the hamburguer and close icon were change from html img
tag to css background image. In order to make easier the javascript code
to toggle them.
+ The new styles were given using the SASS format (scss) in order to
take advantage of the already set of the SASS workflow from the section
of CSS preprocessors

xandro2021 committed:
Add the navigation to the different sections

+ the ancher tags now has the reference to the required sections

xandro2021 committed:
Fix javascript file by pulishing the initial ideas

xandro2021 committed:
Add smooth scroll to the html

xandro2021 committed:
Change normalize.scss for normalize.css

+ It is just to separate that portion of css that was written by others
in order to clean the default styles by the different browsers. It is
because I don't know if the person in charge of the code review will
check directly from the SASS files of from the generated vanilla CSS

xandro2021 committed:
Filter the mobile menu script only for small screens

xandro2021 committed:
fix javascript and Sass linters

ABDULALI3468 reviewed (commented):
Hi @xandro2021 @Ruthmy , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

## Reason for invalidaition:-

**1-** In this project, your README file is not professional. You have added a tech stack that you are not using in the project. I request you consider changing it with the stack you have used to build your project.

### Screenshot:-

I am attaching the screenshot of the tech stack to let you understand clearly what I am taking about:-

![image](https://user-images.githubusercontent.com/94223367/223340970-095e4366-f1f8-4b41-aebf-67f7241536e0.png)

**2-** In your README file please consider removing the `Live Demo` section of your README if you decided not to make it live as it's not a project requirement else consider making the project live and adding a link to the live version of the project to make your repository professional.

![image](https://user-images.githubusercontent.com/94223367/223342973-473d51af-78bf-4c8f-a891-3647dfb4e565.png)



Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

xandro2021 committed:
Remove tech-stacks that are not being used

xandro2021 committed:
Add github pages link as a Live Demo Link

xandro2021 committed:
update readme

xandro2021 commented:
Ok, thank you for the review. I have already updated the readme. As a reminder, my live-version is from the main branch the new changes won't be reflected until I merge this feature branch

leonardodiasb reviewed (changes requested):
# INVALID ❌ ❌ ❌ 

Hi @xandro2021, 👋

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌
 - This is the first project using Javascript and the Javascript linters should be set up correctly. I see that you added the `.eslintrc.json` file, but the `linters.yml` file in the `.github/workflows` folder should be updated too. Please update that file following [these instructions](https://github.com/microverseinc/linters-config/tree/master/html-css-js).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

✅ Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_⚠️ Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

----
_Invalid Code Review Request does not count into the code reviews limit._

xandro2021 committed:
Update linter files

xandro2021 committed:
Previous ruless for SCSS files

+ The new rules are forcing me to use old and obsolete SCSS code. @use
and @forward were new things introduced in SASS because the default
@import used in vanilla SCSS was causing issues, as result the people in
charge of the development of SASS created those alternatives. One has a
similar functionality than import while the other is a link to address
to the file. Similar to pointers in C++.
+ @mixin? Seriously, what's the point of using SASS without mixins? SASS
is something more than just create variable and wrap css rules.

+ @include? yeah, I mean considering that @mixin are givin issue, I'm
having the feeling that the are very outdated and just see SASS as a
tool to wrap selectors.

NOTE: I won't work with outdate SASS. If that is an issue with
Microverse I will take my apportunity of still being in week 04 and drop
the program. As a developer I'm pursing for being update with the tools,
I cannot show a employer a portfolio with SASS ignoring mixins and
includes, while at same time ignoring
using imports instead of the right use of @forward and @use
I DONT WANT to make them thik I see .scss files as an old .sass files with
brackets instead of indentation.

xandro2021 committed:
update readme

xandro2021 committed:
delete old linter

xandro2021 committed:
new empty linter file

xandro2021 committed:
copy the content from Microverse linter yaml file to the project

xandro2021 committed:
change Node Version.

+ The config with node 18.x is trigguering issues despite follow all the
steps from the microverse guide https://github.com/microverseinc/linters-config/tree/master/html-css-js
I follow the guide several times, but the issue persist. Something is
not working in your lighthouse configuration with a newer node version

xandro2021 commented:
# Important NOTES:
## Commit [3a46fc4](https://github.com/MyMicroverseProjects/PortfolioProject/pull/12/commits/3a46fc4822d5ed724a0c1b3ec7882717782df3ae)
Looks like Microverse SCSS rules are the same as the old obsolete SASS, making SCSS files looks just like wrapper of CSS selectors and nothing else. Microverse rules are Ignoring the existence of the progress made in SCSS with @use, @forward, @mixin and @include of the modern SCSS files that part of the modern practices in SASS. More information in the commit.

## Commit [9ec5198](https://github.com/MyMicroverseProjects/PortfolioProject/pull/12/commits/9ec519847a818ef620d18dea9ef649f81013c89f)
You did a huge update of a Node version to another (12.x to 18x), and because of that your current configuration is broken for the Lighthouse case. After wasting almost two hours following your tutorial several times, with some iddle commits because not even git was noticing the differences in my action when I was overwriting the files when I copy and paste or create or download your files from https://github.com/microverseinc/linters-config/tree/master/html-css-js 
The connection was never successful. The only solution was to revert the node version to 12.x only to the LightHouse section like how it was before a few weeks ago.

tresorsawasawa commented on src/scss/header/_navegacion.scss:
```diff
@@ -1,23 +1,60 @@
+    display: block;
+  }
 }
 
-.menu-icon:hover {
-  transform: scale(1.3);
-  transition-duration: 0.3s;
-  color: #6070ff;
+// Trying to create the cover
+.menu.reveal {
+  display: block;
+  position: absolute;
+  height: 100vh;
+  width: 100vw;
+  right: 0;
+  top: 0;
+  background-color: #6070ffed;
+
+  ul {
+    width: 90%;
+    margin: 0 auto;
+    margin-top: 4.8rem;
+
+    .menu-icon {
+      display: block;
+      margin: 0;
+
+      a {
+        color: #fff;
+        font-weight: 600;
+        font-size: 3.2rem;
+      }
+
+      &:hover {
+        background-color: #8291ff;
+        scale: 1.03;
+      }
+    }
+  }
 }
```
- [ ] I think it would be better if you update your mobile menu design in order it follows the design from [your chosen template in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1). Right now, you can see that it misses the `blur effect` on the **background**.

| Figma Design | Your Design |
| -------------------- | ---------------- |
| ![Screenshot from 2023-01-04 13-57-26](https://user-images.githubusercontent.com/83042742/223242457-5b5a8c30-dd8e-4923-878d-4ab4eadb7d13.png) | ![Screenshot from 2023-03-07 18-57-31](https://user-images.githubusercontent.com/83042742/223494774-3a6874a6-a237-4572-9f8f-09f2c33ce115.png) |

tresorsawasawa reviewed (changes requested):
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @xandro2021 and @muneeb3123 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

![almost-there](https://user-images.githubusercontent.com/57562869/143511358-9cde7daf-4ad0-4db2-8f15-18b54288ff91.gif)

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

xandro2021 committed:
Add blur effect in the menu background

huseyinbiyik reviewed (approved):
Hi team,

## ✔️ Status: APPROVED 🏆 🎉 🎊

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![Alt Text](https://media.giphy.com/media/wZjlCH43M3M0U/giphy.gif)



## Highlights
- Design implementation looks perfect 👌 
- Linter set up well and checks pass 🟢
- Github flow followed 👌
- PR description is descriptive 📝
- Great implementation of the requested changes of previous reviewers 🕸️

## Optional suggestions
- Commit messages are perfect but let's capitalize the first word to increase readability 👍 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
